lib/command/src/Obelisk/Command/Project.hs

```diff
@@ -199,20 +205,21 @@ walkToProjectRoot this thisStat mDesiredUmask myUid = liftIO (doesDirectoryExist
 -- | Walk from the given project root directory to its Obelisk implementation
 -- directory, accumulating potentially insecure directories that were traversed
 -- in the process.
-walkToImplDir :: (MonadState [FilePath] m, MonadIO m) => FilePath -> UserID -> m ()
-walkToImplDir projectRoot myUid = do
+walkToImplDir :: (MonadState [FilePath] m, MonadIO m) => FilePath -> UserID -> FileMode -> m ()
+walkToImplDir projectRoot myUid umask = do
   let obDir = toObeliskDir projectRoot
   obDirStat <- liftIO $ getFileStatus obDir
-  when (not $ isWritableOnlyBy obDirStat myUid) $ modify (obDir:)
+  when (not $ isWellOwnedAndWellMasked obDirStat myUid umask) $ modify (obDir:)
   let implThunk = obDir </> "impl"
   implThunkStat <- liftIO $ getFileStatus implThunk
-  when (not $ isWritableOnlyBy implThunkStat myUid) $ modify (implThunk:)
+  when (not $ isWellOwnedAndWellMasked implThunkStat myUid umask) $ modify (implThunk:)
 
---TODO: Is there a better way to ask if anyone else can write things?
---E.g. what about ACLs?
--- | Check to see if directory is only writable by a user whose User ID matches the second argument provided
-isWritableOnlyBy :: FileStatus -> UserID -> Bool
-isWritableOnlyBy s uid = fileOwner s == uid && fileMode s .&. 0o22 == 0
+-- | Check to see if directory is writable by a user whose User ID matches the
+-- second argument provided, and if the fact that other people can write to that
+-- directory is in accordance with the umask of the system, passed as the third
+-- argument.
+isWellOwnedAndWellMasked :: FileStatus -> UserID -> FileMode -> Bool
```

Comment:
"WellMasked" doesn't seem quite right. The mask affects what permissions are assigned by default.  But once the file is created it's no longer using a mask...it just has permissions. Maybe `hasSafePermissions` or something...